src/util/rgbHex.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Converts RGBA color values to a hexadecimal color string.
+ * @returns Hexadecimal color string in the format #RRGGBBAA.
+ */
+export const rgbaToHex = (rgba: string): string => {
+  // Extract the RGBA components from the string using a regular expression
+  const rgbaRegex = /^rgba?\((\d+),\s*(\d+),\s*(\d+),\s*([01]?\.?\d*)\)$/
+  const match = rgba.match(rgbaRegex)
+
+  if (!match) {
+    throw new Error('Invalid RGBA color format')
+  }
+
+  // Parse the extracted components
+  const r = parseInt(match[1])
+  const g = parseInt(match[2])
+  const b = parseInt(match[3])
+  const a = parseFloat(match[4])
+
+  /** Convert RGB to hex. */
+  const toHex = (value: number) => Math.min(Math.max(value, 0), 255).toString(16).padStart(2, '0')
+
+  if (a >= 1) {
+    return `#${toHex(r)}${toHex(g)}${toHex(b)}` // RGB hex only
+  } else {
+    const alphaHex = Math.round(a * 255)
+      .toString(16)
+      .padStart(2, '0')
+    return `#${toHex(r)}${toHex(g)}${toHex(b)}${alphaHex}` // RGBA hex
+  }
```

Comment:
A slight improvement is to move the alpha condition into `alphaHex` itself. This will avoid duplicating the return statement and string interpolation.

```ts
const alphaHex = a >= 1 ? Math.round(a * 255)
      .toString(16)
      .padStart(2, '0') : ''
return `#${toHex(r)}${toHex(g)}${toHex(b)}${alphaHex}`
```